src/utf8.rs

```diff
@@ -84,8 +84,8 @@ impl Validator {
             unsafe {
                 self.partial_codepoint
                     .get_unchecked_mut(self.partial_codepoint_len..codepoint_len_after_copy)
-                    .copy_from_slice(input.get_unchecked(..bytes_to_copy));
             }
+            .copy_from_slice(&input[..bytes_to_copy]);
```

Comment:
`bytes_to_copy` is computed above based on `input.len()`

Reply:
This is fine, see https://godbolt.org/z/zf64s7KxG